### DISCUSSION TITLE
Would it be possible to set a layout for specific folders of pages?

### DISCUSSION THREAD

**Discussion by robertt** [vercel/next.js#13630] [Help] [answered]
As the title says, is there any way I can define a file that's within a folder of pages, that gets passed a children prop with the content of the current child page. For example, lets say I have a `settings` folder. I want the settings folder to render some global stuff (tabs for instance), when you go to `settings/collaborators` it passes the content of the `collaborators` page to the parent page.

Hope that makes sense and thanks in advance!

**Comment by rafaelalmeidatk** [comment, marked as answer]
This is not possible yet, you can track this issue: https://github.com/vercel/next.js/issues/10361
The solution now is to create layout components (like `SettingsLayout`) and compose your page components with these components:

```js
const SettingsCollaboratorsPage = () => (
  <SettingsLayout>
    ...
  </SettingsLayout>
);
```

**Comment by Vadorequest** [comment]
The way I do that is by allowing the page to define which layout it should use.

https://github.com/UnlyEd/next-right-now/blob/df72481c54/src/pages/%5Blocale%5D/examples/index.tsx#L53-L59

But that's not automated. But it allows me to do it per folder, manually.
